app/gosns/set_subscription_attributes.go

```diff
@@ -0,0 +1,74 @@
+package gosns
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func SetSubscriptionAttributesV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+	requestBody := models.NewSetSubscriptionAttributesRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+	if !ok {
+		log.Error("Invalid Request - SetSubscriptionAttributesV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+
+	subsArn := requestBody.SubscriptionArn
+	attrName := requestBody.AttributeName
+	attrValue := requestBody.AttributeValue
+
+	sub := getSubscription(subsArn)
+	if sub == nil {
+		return utils.CreateErrorResponseV1("SubscriptionNotFound", false)
+	}
+
+	switch attrName {
+	case "RawMessageDelivery":
+		app.SyncTopics.Lock()
+		if attrValue == "true" {
+			sub.Raw = true
+		} else {
+			sub.Raw = false
+		}
+		app.SyncTopics.Unlock()
+
+	case "FilterPolicy":
+		filterPolicy := &app.FilterPolicy{}
+		err := json.Unmarshal([]byte(attrValue), filterPolicy)
+		if err != nil {
+			return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+		}
+		app.SyncTopics.Lock()
+		sub.FilterPolicy = filterPolicy
+		app.SyncTopics.Unlock()
+
+	default:
+		// FIXME: This response is literally strange but it's the same behavior with old SetSubscriptionAtteributes.
+		return utils.CreateErrorResponseV1("SubscriptionNotFound", false)
+	}
+
+	uuid := uuid.NewString()
+	respStruct := models.SetSubscriptionAttributesResponse{
+		Xmlns:    models.BASE_XMLNS,
+		Metadata: app.ResponseMetadata{RequestId: uuid}}
+
+	return http.StatusOK, respStruct
+}
+
+func getSubscription(subsArn string) *app.Subscription {
+	for _, topic := range app.SyncTopics.Topics {
+		for _, sub := range topic.Subscriptions {
+			if sub.SubscriptionArn == subsArn {
+				return sub
+			}
+		}
+	}
+	return nil
+}
```

Comment:
I tried to separate the loop indent from the following logic.

Reply:
Nice!  If you want to throw it in the utils or somewhere more general it might be helpful in the other handlers?

Reply:
OK, I will apply the idea in this PR. At lease, I will apply it on GetSubscriptionAttributes API.

Reply:
I shared the `getSubscription` to just only GetSubscriptionAttributes API. 
Unsubscribe API logic also has the same loop and indent but the API requires the subscription index in a topic and modify them by topic level perspective. I thought it will be too much commonize for the API.